### PR TITLE
Feature: Capture Activity Data 

### PR DESCRIPTION
**Pull request by alexbielen**
**Demo**:
Dec-04-2019 14-38-06

Add the Capture Activity Data modal and send fields to the backend. 

ActivityForm schema looks like the following: 
```js
const activityFormValues = {
  activityType: ACTIVITY_TYPES[0], // String 
  description: undefined, // String
  dateCompleted: new Date(), // DateTime
  timeSpentInMinutes: TIME_SPENT_TYPE[0].value, // Int
  difficultyLevel: DIFFICULTY_LEVEL[0], // String
  activityFeeling: [], // [String?]
  whyIfeelThisWay: null, // String? 
};
```

Note that we added a new component `ToggleButton` to get around some of the limitations with MUI ButtonGroups.

### CODE REVIEW

File: components/dashboard/ActivityInputDialog.js [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                    'aria-label': 'change date',
+                  }}
+                />
+              </Grid>
+            </MuiPickersUtilsProvider>
+            <FormControl className={classes.formControl}>
+              <InputLabel id={`${formId}-timeSpent`}>Time Spent</InputLabel>
+              <Select
+                labelId={`${formId}-timeSpent`}
+                id="timeSpent-select"
```
Comment (rossdakin) [inline components/dashboard/ActivityInputDialog.js]: Could we include `formId` in this (and a few others), just to keep the IDs "namespaced"?

Also note the default `customized-dialog-title`

File: components/dashboard/ActivityInputDialog.js [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+  const resetComponent = () => {
+    setError();
+    setSuccess(false);
+    setSubmitting(false);
+    setFormValues(activityFormValues);
+    setFormErrors({});
+  };
+
+  return (
+    <Dialog
```
Comment (rossdakin) [inline components/dashboard/ActivityInputDialog.js]: Thoughts on using `withMobileDialog` like the existing dialogs to go full-screen in mobile?

File: components/dashboard/ActivityInputDialog.js [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+            </FormControl>
+            <MuiPickersUtilsProvider utils={DateFnsUtils}>
+              <Grid container className={classes.formControl}>
+                <KeyboardDatePicker
+                  disableToolbar
+                  disableFuture
+                  variant="inline"
+                  format="MM/dd/yyyy"
+                  margin="normal"
+                  id={`${formId}--dateCompleted`}
```
Comment (rossdakin) [inline components/dashboard/ActivityInputDialog.js]: Nit: could be single dash for consistency.

File: components/dashboard/ActivityInputDialog.js [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+          </form>
+        )}
+        {error && (
+          <Typography color="error" variant="h4">
+            Error: {error}
+          </Typography>
+        )}
+        {success && (
+          <Grid container direction="column" justify="center" alignItems="center">
+            <CheckCircleIcon style={{ fontSize: 200, color: 'green' }}> </CheckCircleIcon>
```
Comment (rossdakin) [inline components/dashboard/ActivityInputDialog.js]: Nit: could be self-closing tag.

**Comment by rossdakin** [comment]
Also getting a couple console errors:

```
index.js:1 Warning: Failed prop type: Invalid prop `value` supplied to `ToggleButton`.
    in ToggleButton (at ActivityInputDialog.js:306)
    in ActivityInputDialog (at Dashboard.js:182)
    in div (at Dashboard.js:181)
    in Dashboard (at dashboard.js:30)
    in DashboardPage (at withTitle.js:14)
    in Unknown (at withAuthRequired.js:6)
    in AuthRequired (at withAuthRequired.js:5)
    in Unknown (at _app.js:110)
    in main (at AppManager.js:112)
    in AppManager (at _app.js:109)
    in AuthProvider (at _app.js:107)
    in FirebaseProvider (at _app.js:106)
    in SnackbarProvider (at _app.js:105)
    in ThemeProvider (at _app.js:104)
    in Container (at _app.js:100)
    in MyApp
    in Suspense
```
```
index.js:1 Warning: Failed prop type: The prop `xs` of `Grid` must be used on `item`.
    in WithStyles(ForwardRef(Grid)) (at ActivityInputDialog.js:298)
    in ActivityInputDialog (at Dashboard.js:182)
    in div (at Dashboard.js:181)
    in Dashboard (at dashboard.js:30)
    in DashboardPage (at withTitle.js:14)
    in Unknown (at withAuthRequired.js:6)
    in AuthRequired (at withAuthRequired.js:5)
    in Unknown (at _app.js:110)
    in main (at AppManager.js:112)
    in AppManager (at _app.js:109)
    in AuthProvider (at _app.js:107)
    in FirebaseProvider (at _app.js:106)
    in SnackbarProvider (at _app.js:105)
    in ThemeProvider (at _app.js:104)
    in Container (at _app.js:100)
    in MyApp
    in Suspense
```

File: package-lock.json [hunk 5]
```diff
@@ -9957,22 +9991,19 @@
           "version": "3.0.4",
           "resolved": "https://registry.npmjs.org/minimatch/-/minimatch-3.0.4.tgz",
           "integrity": "sha512-yJHVQEhyqPLUTgt9B83PXu6W3rx4MvvHvSUvToogpwoGDOUQ+yDrR0HRot+yOCdCO7u4hX3pWft6kWBBcqh0UA==",
-          "optional": true,
```
[the remark this answers is not in the document]
Comment (alexbielen) [reply package-lock.json]: Same. Committing after running `npm i`.

File: components/dashboard/ActivityInputDialog.js (hunk 2 above)
Comment (alexbielen) [reply components/dashboard/ActivityInputDialog.js]: Added.

File: components/dashboard/ActivityInputDialog.js (hunk 4 above)
Comment (alexbielen) [reply components/dashboard/ActivityInputDialog.js]: Good call! Changed.

File: components/dashboard/ActivityInputDialog.js [hunk 6]
```diff
@@ -298,6 +298,7 @@ function ActivityInputDialog({ show, onClose }) {
               <Grid
                 container
                 xs={12}
+                item
```
Comment (rossdakin) [inline components/dashboard/ActivityInputDialog.js]: Feels odd for this to be a container and an item, but nbd if it works.

**Pull request merged**